### ISSUE TITLE
Integration builds takes lots of time

### ISSUE THREAD

**Issue by kameshsampath** [apache/camel-k#1209]
**Platform**: minikube 1.6.2
**camel-k**: 1.0.0.RC1

The camel-k integration builds takes lots of time to complete ~ 12 mins and i see its struck in a particular place for long  as shown in the log below ( not sure its Kankio issue)

```shell
camel-k-kit-bofk6gui1aal47gim840-builder kaniko INFO[0492] Downloading base image 10.111.197.108/chapter-7/camel-k-kit-bofj39ei1aal47gim830:36212
camel-k-kit-bofk6gui1aal47gim840-builder kaniko INFO[0647] Skipping unpacking as no commands require it.
camel-k-kit-bofk6gui1aal47gim840-builder kaniko INFO[0647] Taking snapshot of full filesystem...
camel-k-kit-bofk6gui1aal47gim840-builder kaniko INFO[0647] Using files from context: [/builder/kit-bofk6gui1aal47gim840/package/context]
camel-k-kit-bofk6gui1aal47gim840-builder kaniko INFO[0647] ADD . /deployments
camel-k-kit-bofk6gui1aal47gim840-builder kaniko INFO[0647] Taking snapshot of files...
```
Is there a way to optimize this ? As majority of developers will be on minikube we need to make sure its faster and quicker there.

**Comment by astefanutti** [comment]
@kameshsampath out of curiosity, what OS do you use?

Kaniko takes a snapshot of the file system and performance may depend on the OS: https://github.com/GoogleContainerTools/kaniko/blob/200a0eabdd4669c6bcacb948e1b3f52b56fd1f5f/pkg/snapshot/snapshot.go#L122-L125.

**Comment by kameshsampath** [comment]
@astefanutti  - I use macOS Catalina

**Comment by astefanutti** [comment]
@kameshsampath that's what I suspected. As Minikube runs through virtualisation layers on Mac OS, it may be the file system snapshotting is particularly slow for that platform. I'd like to hear from people using Linux if they are also affected.

**Comment by astefanutti** [comment]
I had a quick look at it and it seems a significant portion of the build time is taken to push the built image to the internal registry. Still need to be confirmed and properly analysed...

**Comment by astefanutti** [comment]
After a deeper analysis, it seems to confirm most of the time is taken by the image upload to the local registry, I've created GoogleContainerTools/kaniko#1013 to track progress on this.

**Comment by astefanutti** [comment]
It should be fixed with #1256.
